### PR TITLE
ENH: Support transformix_filter functional interface

### LINKED ISSUE #69: Procedural interface Transformix not working in 3D
The procedural interface of transformix is not working in 3D, see example notebook 9.
For transformix to work in 3D in the object oriented interface an explicit call to the 3D transformix class is necessary

### PR DESCRIPTION
**Pull request by thewtex**
Closes #69.

### CODE REVIEW

**Comment by thewtex** [comment]
Closes #69 

elastix PR: https://github.com/SuperElastix/elastix/pull/323

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,8 +40,10 @@ set(_itk_build_testing ${BUILD_TESTING})
 set(_itk_build_shared ${BUILD_SHARED_LIBS})
 set(BUILD_SHARED_LIBS OFF)
 set(elastix_GIT_REPOSITORY "https://github.com/SuperElastix/elastix.git")
+set(elastix_GIT_REPOSITORY "https://github.com/thewtex/elastix.git")
 # Tag of commit "COMP: Quick-fix ITKElastix error: 'itkCostFunction.h' file not found"
```
Comment (N-Dekker) [inline CMakeLists.txt:44]: @thewtex Can you please keep this line of comment "# Tag of commit "COMP..." in sync with the `elastix_GIT_TAG`? (Or just remove the line of comment, if you think it is too cumbersome to keep it up-to-date.)

Comment (thewtex) [reply CMakeLists.txt:44]: Yes, removed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -40,8 +40,9 @@ set(_itk_build_testing ${BUILD_TESTING})
 set(_itk_build_shared ${BUILD_SHARED_LIBS})
 set(BUILD_SHARED_LIBS OFF)
 set(elastix_GIT_REPOSITORY "https://github.com/SuperElastix/elastix.git")
-# Tag of commit "COMP: Quick-fix ITKElastix error: 'itkCostFunction.h' file not found"
+set(elastix_GIT_REPOSITORY "https://github.com/thewtex/elastix.git")
```
Comment (dzenanz) [inline CMakeLists.txt:43]: I guess that second `set` call has effect (and keep the first one for easy reverting later), but this looks suspiciously like a rebase gone wrong.

Comment (thewtex) [reply CMakeLists.txt:43]: Since this has not been merged upstream, yet, we have to use the repository where it exists.

**Comment by thewtex** [comment]
@N-Dekker are the Linux build failures,

```
In file included from _deps/elx-src/Common/xout/xoutsimple.h:76:0,
                 from _deps/elx-src/Common/xout/xoutmain.h:23,
                 from _deps/elx-src/Core/Configuration/elxConfiguration.h:27,
                 from _deps/elx-src/Core/Kernel/elxElastixBase.h:33,
                 from _deps/elx-src/Core/Kernel/elxElastixTemplate.h:21,
                 from _deps/elx-src/Core/Install/elxIncludes.h:39,
                 from _deps/elx-src/Components/FixedImagePyramids/FixedGenericPyramid/elxFixedGenericPyramid.h:21,
                 from _deps/elx-src/Components/FixedImagePyramids/FixedGenericPyramid/elxFixedGenericPyramid.cxx:19:
_deps/elx-src/Common/xout/xoutsimple.hxx:98:3: error: prototype for ‘const typename xoutlibrary::xoutsimple<charT, traits>::XStreamMapType& xoutlibrary::xoutsimple<charT, traits>::GetXOutputs()’ does not match any in class ‘xoutlibrary::xoutsimple<charT, traits>’
 & xoutsimple< charT, traits >::GetXOutputs( void )
   ^
In file included from _deps/elx-src/Common/xout/xoutmain.h:23:0,
                 from _deps/elx-src/Core/Configuration/elxConfiguration.h:27,
                 from _deps/elx-src/Core/Kernel/elxElastixBase.h:33,
                 from _deps/elx-src/Core/Kernel/elxElastixTemplate.h:21,
                 from _deps/elx-src/Core/Install/elxIncludes.h:39,
                 from _deps/elx-src/Components/FixedImagePyramids/FixedGenericPyramid/elxFixedGenericPyramid.h:21,
                 from _deps/elx-src/Components/FixedImagePyramids/FixedGenericPyramid/elxFixedGenericPyramid.cxx:19:
_deps/elx-src/Common/xout/xoutsimple.h:70:26: error: candidate is: const typename xoutlibrary::xoutsimple<charT, traits>::Superclass::XStreamMapType& xoutlibrary::xoutsimple<charT, traits>::GetXOutputs()
   const XStreamMapType & GetXOutputs( void ) override;
                          ^
In file included from _deps/elx-src/Common/xout/xoutsimple.h:76:0,
                 from _deps/elx-src/Common/xout/xoutmain.h:23,
                 from _deps/elx-src/Core/Configuration/elxConfiguration.h:27,
                 from _deps/elx-src/Core/Kernel/elxElastixBase.h:33,
                 from _deps/elx-src/Core/Kernel/elxElastixTemplate.h:21,
                 from _deps/elx-src/Core/Install/elxIncludes.h:39,
                 from _deps/elx-src/Components/FixedImagePyramids/FixedGenericPyramid/elxFixedGenericPyramid.h:21,
                 from _deps/elx-src/Components/FixedImagePyramids/FixedGenericPyramid/elxFixedGenericPyramid.cxx:19:
_deps/elx-src/Common/xout/xoutsimple.hxx:110:3: error: prototype for ‘const typename xoutlibrary::xoutsimple<charT, traits>::CStreamMapType& xoutlibrary::xoutsimple<charT, traits>::GetCOutputs()’ does not match any in class ‘xoutlibrary::xoutsimple<charT, traits>’
 & xoutsimple< charT, traits >::GetCOutputs( void )
   ^
```

related to recent changes on the `elastix` `develop` branch?

**Review by dzenanz — COMMENTED** [review]
LGTM. Point to official repo before merging.

**Comment by thewtex** [comment]
> related to recent changes on the elastix develop branch?

This appears to be the case -- I added the patches in https://github.com/SuperElastix/elastix/pull/323 before the xout changes on the `develop` branch, and the builds are now passing. Tracked in https://github.com/SuperElastix/elastix/issues/331

**Review by ViktorvdValk — APPROVED** [review]
Nice, looks good!
I'll update the example notebooks after the merge.

**Comment by thewtex** [comment]
Updating the `elastix` version to current `develop` that incorporates https://github.com/SuperElastix/elastix/pull/323 and @N-Dekker xout fixes in https://github.com/SuperElastix/elastix/pull/330.

**Pull request merged**